wolfcrypt/src/asn.c

```diff
@@ -26227,8 +26227,10 @@ static int SetReqAttribSingle(byte* output, int* idx, char* attr, int attrSz,
         if (strSz > 0) {
             XMEMCPY(&output[*idx], str, strSz);
             *idx += strSz;
-            XMEMCPY(&output[*idx], attr, attrSz);
-            *idx += attrSz;
+            if (attrSz > 0) {
```

Comment:
`memcpy` should be able to handle a 0 size arg. I'm positive this is not the only place we make that assumption.

Reply:
AFAIK it is indeed benign, but in the header file from glibc it's ```extern void *memcpy (void *__restrict __dest, const void *__restrict __src,
                     size_t __n) __THROW __nonnull ((1, 2));
``` so it is simply forbidden to pass a known-null to it.  so `clang-tidy` is right.

the codebase no longer has any other examples of known-nulls being passed as a `__nonnull` arg -- I fixed various of them a while back while making the codebase `clang-tidy`-compatible.